pkg/collector/corechecks/snmp/internal/checkconfig/config.go

```diff
@@ -431,14 +439,33 @@ func NewCheckConfig(rawInstance integration.Data, rawInitConfig integration.Data
 	}
 
 	// SNMP connection configs
-	c.CommunityString = instance.CommunityString
-	c.User = instance.User
-	c.AuthProtocol = instance.AuthProtocol
-	c.AuthKey = instance.AuthKey
-	c.PrivProtocol = instance.PrivProtocol
-	c.PrivKey = instance.PrivKey
-	c.ContextName = instance.ContextName
+	if instance.CommunityString != "" {
```

Comment:
At the moment, the logic that create `gosnmp.GoSNMP` also happen in two other places:

- https://github.com/DataDog/datadog-agent/blob/1e872f51c70d105f9b345938be8019a525a29ce8/pkg/snmp/snmp.go#L203-L257
- https://github.com/DataDog/datadog-agent/blob/e6688b3e605118b21c3a2b0638267d532ff2ad19/cmd/agent/subcommands/snmp/command.go#L237-L241

we should ideally factor the logic to avoid duplication/repetition.

Right now, optionally, you might want to add the same default proto logic for md5/des at those 2 places.